code/mob/living/critter/ai/generic_critter.dm

```diff
@@ -207,6 +207,84 @@
 /datum/aiTask/succeedable/critter/eat/on_reset()
 	has_started = FALSE
 
+
+/// This one makes the mob move towards a target mob and attack it. Repeats until the target is dead, gone, too far, or we are incapacitated. Called upon being attacked if the ai is set to retaliate
+/datum/aiTask/sequence/goalbased/retaliate
+	name = "retaliating"
+	weight = -100
+	max_dist = 7
+	ai_turbo = TRUE
+	var/mob/targetted_mob = null
+	var/start_time = 0
+
+
+/datum/aiTask/sequence/goalbased/retaliate/New(parentHolder, transTask)
+	..()
+	add_task(holder.get_instance(/datum/aiTask/succeedable/retaliate, list(holder)))
+
+/datum/aiTask/sequence/goalbased/retaliate/get_targets()
+	return list(src.targetted_mob)
+
+////////
+
+/datum/aiTask/succeedable/retaliate
+	name = "retaliate subtask"
+	max_dist = 7
+	var/has_started = FALSE
+	var/persistence = 0
+
+
+/datum/aiTask/succeedable/retaliate/failed()
+	var/mob/living/critter/C = holder.owner
+	var/mob/T = holder.target
```

Comment:
nonstandard single letter var names!!1!

Reply:
removed the vars, cos I wasn't actually using them